components/external-services-mock/internal/systemfetcher/handler.go

```diff
@@ -15,13 +15,14 @@ import (
 type SystemFetcherHandler struct {
 	mutex         sync.Mutex
 	defaulTenant  string
-	mockedSystems []byte
+	mockedSystems [][]byte
```

Comment:
What is the purpose of those `mockedSystems`?

Reply:
**mockedSystems** stores the systems which the external-services-mock will return on calls to `/systemfetcher/fetch`. It serves as a queue - on each call the first system that has been mocked through the `/systemfetcher/configure` endpoint will get returned. In order to be able to test paging functionality, we need to configure more than one response (at least 2 pages). That is why it is a slice of slices now.

Reply:
Oh, sorry, I've misread the location of this file. I didn't see it was in the `ext-svc-mock` service. I though it was part of the system fetcher service itself which then would've been a concern. It's all good! 👍🏼